source/MaterialXCore/Element.h

```diff
@@ -445,6 +445,20 @@ class MX_CORE_API Element : public std::enable_shared_from_this<Element>
         return child ? child->asA<T>() : shared_ptr<T>();
     }
 
+    /// Return the child element from data library , if any, with the given name and subclass.
+    /// If a child with the given name exists, but belongs to a different
+    /// subclass, then an empty shared pointer is returned.
+    template <class T> shared_ptr<T> getChildOfType(ConstElementPtr datalibrary, const string& name) const
```

Comment:
@ashwinbhat I'd recommend leaving the interface of these core methods, `getChildOfType` and `getChildrenOfType` unchanged in your PR, and creating a brand new name for the methods that you'd like to add.  This should allow us to focus on just the C++ API in this first phase of review, leaving the authoring of Python and JavaScript bindings to a future phase.

Reply:
@ashwinbhat That sounds fine to start with, and I suspect many of these helper methods will evolve during the review and discussion process!